libvast/src/system/index.cpp

```diff
@@ -436,6 +436,12 @@ caf::behavior index(caf::stateful_actor<index_state>* self, const path& dir,
   self->set_exit_handler([=](const caf::exit_msg& msg) {
     VAST_DEBUG(self, "received exit from", msg.source,
                "with reason:", msg.reason);
+    auto& st = self->state;
+    if (!st.unpersisted.empty() || st.active_partition_indexers > 0) {
+      auto delay = defaults::index::shutdown_retry_interval;
+      VAST_INFO(self, "delaying exit by", delay, "to wait for outstanding indexers");
+      self->delayed_send(self, delay, msg);
+    }
```

Comment:
Don't we also need to tell the indexers to exit when we get here for the first time?

Reply:
Good question, I was assuming this only happens when the system shuts down so everything else  should have gotten exit messages as well.

If we can also shutdown the index at runtime then I think we *do* need to kill the indexers here as well (and add an API to `partition` for that, since the index doesnt have direct knowledge of the existing indexers)

I'm wondering if it could happen that a new indexer is about to be launched but it doesnt exist yet when the `exit` message arrives at the index.